scripts/Utils.py

```diff
@@ -75,12 +77,13 @@ def DownloadFile(url, filepath):
                     avgKBPerSecond = (downloaded / 1024) / elapsedTime
                 except ZeroDivisionError:
                     avgKBPerSecond = 0.0
-
+                estimatedTime = ( (total - downloaded) / (avgKBPerSecond*1024) )
+                estimatedTimeString = FormatTime(estimatedTime)
                 avgSpeedString = '{:.2f} KB/s'.format(avgKBPerSecond)
```

Comment:
For flow of the program, quite a small detail, but I would suggest putting the speed first and then the time, as we first calculated the speed as well...
```suggestion
                avgSpeedString = '{:.2f} KB/s'.format(avgKBPerSecond)
                estimatedTimeString = FormatTime(estimatedTime)
```